### PR TITLE
Removed open prop from shouldForwardProp in Drawer root

### LINKED ISSUE #486: Temporary Drawer variant not working on Dev docs
#### Describe the bug
Drawer variant `temporary` not working inside Drawer docs playground
#### What is the expected behaviour?
Drawer variant `temporary` should open overlay.
#### What are the steps to reproduce?
1. Go to /components/drawer/playground
2. Select a **Temporary** variant from dropdown
3. Observe the playground it hides the drawer.

#### Screenshots or links to minimum reproduction example
Temporary2

#### Environment

		All

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes #486 

#### Changes proposed in this Pull Request:
- Removed open prop from shouldForwardProp in Drawer Root

#### Screenshots:
**Before**
Screenshot 2022-08-30 at 2 58 08 PM

**After**
Screenshot 2022-08-30 at 3 08 10 PM

#### To Test:
- yarn start:showcase
**Note: temporary variant of drawer by default takes open prop as false, so to test this we need to pass open prop in Drawer Root element as true and in NavigationDrawer we need to change the variant to temporary **

### CODE REVIEW

**Comment by ektasawant** [comment]
Please update changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 -   Components are now styled using use the new MUI styled engine and extend the sx prop.
 
+### Fixed
```
Comment (emclaug2) [inline CHANGELOG.md:9]: You need a new changelog entry for this.  v6.1.1

**Comment by emclaug2** [comment]
Is this something that we want to publish quickly? If not it may make sense to merge this into the release branch instead.

**Comment by ektasawant** [comment]
> Is this something that we want to publish quickly? If not it may make sense to merge this into the release branch instead.

Can we merge it into the release branch? There are other bugs we can find in react component library while developing dev docs.

**Comment by bkarambe** [comment]
@emclaug2 @ektasawant Updated the base branch.

**Pull request merged**